paramak/utils.py

```diff
@@ -456,34 +456,168 @@ def plotly_trace(
     return trace
 
 
-def export_html_stp_file():
+def extract_points_from_edges(
+    edges,
+    view_plane: str = 'XZ',
+):
+
+    if isinstance(edges, list):
+        list_of_edges = edges
+    else:
+        list_of_edges = [edges]
+
+    points = []
+
+    for edge in list_of_edges:
+        for vertex in edge.Vertices():
+            if view_plane == 'XZ':
+                points.append((vertex.X, vertex.Z))
+            elif view_plane == 'XY':
+                points.append((vertex.X, vertex.Y))
+            elif view_plane == 'YZ':
+                points.append((vertex.Y, vertex.Z))
+            elif view_plane == 'YX':
+                points.append((vertex.Y, vertex.X))
+            elif view_plane == 'ZY':
+                points.append((vertex.Z, vertex.Y))
+            elif view_plane == 'ZX':
+                points.append((vertex.Z, vertex.X))
+            elif view_plane == 'RZ':
+                xy = math.pow(vertex.X, 2) + math.pow(vertex.Y, 2)
+                points.append((math.sqrt(xy), vertex.Z))
+            else:
+                raise ValueError('view_plane value of ', view_plane,
+                                 ' is not supported')
```

Comment:
This can be refactored so that it's more generic. What do you think about something like that ?

```suggestion
    for edge in list_of_edges:
        for vertex in edge.Vertices():
            correspondance_table = {
                "X": vertex.X,
                "Y": vertex.Y,
                "Z": vertex.Z,
                "R": math.sqrt(math.pow(vertex.X, 2) + math.pow(vertex.Y, 2))
            }
            vertices = []
            for edge in list_of_edges:
                for ax in view_plane:
                    if ax not in correspondance_table.keys():
                        ValueError('view_plane value of ', view_plane,
                                   ' is not supported')
                    vertices.append(correspondance_table[ax])
                points.append((*vertices))
```

Reply:
that is much more elegant

Reply:
This is nice in terms of the eligance but unfortunately it has a few issues

it is not raising a value error when the view_plane is incorrect.
It is also doing a math.sqrt and math.pow for every vertex even if it is not needed. 
The points.append() is not add a tuple but actually trying to append a couple of items (which fails)

So I think I will switch back to the old method, which is perhaps a little easier to understand. Perhaps this is an upgrade for another time